spec/rspec/rails/matchers/active_job_spec.rb

```diff
@@ -262,6 +262,18 @@ def perform; raise StandardError; end
       }.to have_enqueued_job.on_queue(:low)
     end
 
+    it "passes with provided priority number as integer" do
+      expect {
+        hello_job.set(priority: 2).perform_later
+      }.to have_enqueued_job.with_priority(2)
+    end
+
+    it "passes with provided priority number as string" do
+      expect {
+        hello_job.set(priority: 2).perform_later
+      }.to have_enqueued_job.with_priority("2")
+    end
+
```

Comment:
Can you please add a fee more cases? I can think of a failure case when we expect priority and it wasn’t set. Also when it was set, but doesn’t match the expectation. In both examples with checks if our failure message is reasonable.
I had another scenario on the tip of my tongue, but it keeps slipping. Can you think of other cases we should cover? You can use other similar qualifiers as a source of inspiration .

By the way, does this apply to mailers?